lldb/include/lldb/Target/DynamicRegisterInfo.h

```diff
@@ -93,6 +93,10 @@ class DynamicRegisterInfo {
     return llvm::iterator_range<reg_collection::const_iterator>(m_regs);
   }
 
+  llvm::iterator_range<reg_collection::iterator> registers_mutable() {
```

Comment:
Can't this be just an overload of the `registers()` method above ?

Reply:
Wouldn't that have to overload on return type? I didn't think that was possible.

Reply:
Still not sure here, I guess you could pass a reference to an iterator to be set, then overload on the type of that?

Reply:
You should be able to use template specialization on the return type to use the same method name.

I don't feel strongly about implementing it, if you don't feel comfortable writing templated code, just a nice to have :).

Reply:
That's new to me, so I'll give it a go and worst case I learn something new.